internal/apps/dop/providers/projectpipeline/service_impl.go

```diff
@@ -473,16 +488,52 @@ func (p *ProjectPipelineService) createCronIfNotExist(definition *dpb.PipelineDe
 		return nil
 	}
 
-	createV2 := extra.CreateRequest
-	createV2.DefinitionID = definition.ID
-	_, err = p.pipelineService.PipelineCreateV2(apis.WithInternalClientContext(context.Background(), discover.DOP()), createV2)
+	cronCreateRequest, err := p.constructCronCreateRequestFromV2(createV2, pipelineYml)
+	if err != nil {
+		return fmt.Errorf("failed to constructCronCreateRequestFromV2,err :%v", err)
+	}
+
+	_, err = p.PipelineCron.CronCreate(apis.WithInternalClientContext(context.Background(), discover.DOP()), cronCreateRequest)
```

Comment:
check if we need use context with user id

Reply:
The `checkRolePermission` method is called before creation